### PR TITLE
minor addition to add length for nodeID verification

### PR DESCRIPTION
**Pull request by jacktrusler**
# Description

Added a error message if the Node Id length is not 40 characters. This includes the seven characters 'NodeID-'

## Type of Change

- [ x ] Bug fix
- [ x ] New feature
- [ ] Refactoring
- [ ] Requires a documentation update

# Checklist:

- [ x ] Code duplication check
- [ x ] Positive scenario check - I have performed a self-review of my own code
- [ x ] Unused code cleanup
- [ x ] Clean console
- [ x ] Commented code, particularly in hard-to-understand areas
- [ ] I have created the storybook of the component
- [ ] I have created the unit test of the component

# Video Recording/Screenshots Showcasing Implementation
image

### CODE REVIEW

**Comment by emersoncloud** [comment]
How's this red exclamation mark connected to the error message below? 

image

I put in a valid node id, went to the next page, back to the previous page and the red exclamation mark is there without an error message on that previously valid ID

**Comment by emersoncloud** [comment]
John mentioned yesterday that we can verify the node id with a checksum rather than checking length

**Comment by emersoncloud** [comment]
Alright I added the checksum verification and fixed that right side icon. See what you think @jacktrusler

File: src/modules/components/Wizard/steps/WizardNodeID.tsx [hunk 1]
```diff
@@ -99,12 +81,12 @@ export const WizardNodeID: FunctionComponent<WizardNodeIDProps> = ({
...
-    if (!validNodeId) {
+    if (errorMessage != '') {
       return <WarningIcon color="error.500" />
     }
+    if (errorMessage == '') {
+      return <CheckCircleIcon color="success.500" />
+    }
 
     return null
   }
```
Comment (jacktrusler) [inline src/modules/components/Wizard/steps/WizardNodeID.tsx:92]: Could we make this so the null cases are all removed and just have the function return null if nothing is caught?

File: src/utils/consts.ts [hunk 2]
```diff
@@ -57,3 +57,5 @@ export const DECODED_ERRORS = {
   '0x32d971dc': 'ZeroAssets()',
   '0x9811e0c7': 'ZeroShares()',
 }
+
+export const NODE_ID_LEN = 40
```
Comment (jacktrusler) [inline src/utils/consts.ts:61]: Might not need this anymore

File: src/utils/nodeIdErrorMessage.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import Minipool from '@/types/minipool'
+import { nodeIDToHex } from '@/utils'
+
+export default function nodeIdErrorMessage(minipool: Minipool, debouncedNodeId: string): string {
+  let message = ''
+  try {
+    nodeIDToHex(debouncedNodeId)
+  } catch (e) {
+    console.log("cant' do it man", e)
```
Comment (jacktrusler) [inline src/utils/nodeIdErrorMessage.ts:11]: console.log() he says :v

File: src/utils/nodeIdErrorMessage.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    message = 'NodeID checksum invalid'
+  }
+  if (minipool && !(minipool.status.toNumber() == 4 || minipool.status.toNumber() == 5)) {
+    message = 'NodeID is already in use'
+  }
+  if (debouncedNodeId.length !== NODE_ID_LEN) {
+    message = 'NodeID must be 40 characters'
+  }
```
Comment (jacktrusler) [inline src/utils/nodeIdErrorMessage.ts:19]: Might not need anymore also might not need the starts with node ID check? But maybe it's good to keep to give the user more info.

File: src/utils/nodeIdErrorMessage.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  if (!debouncedNodeId.startsWith('NodeID-')) {
+    message = "NodeID must start with 'NodeID-'"
+  }
+  return message
+}
```
Comment (jacktrusler) [inline src/utils/nodeIdErrorMessage.ts:24]: Might not need anymore also might not need the length? But I think it's good to keep to give the user more info while they're inputting the node ID

File: src/utils/nodeIdErrorMessage.ts (hunk 3 above)
Comment (emersoncloud) [reply src/utils/nodeIdErrorMessage.ts:11]: Lol. How'd that sneak in there

File: src/utils/nodeIdErrorMessage.ts (hunk 5 above)
Comment (emersoncloud) [reply src/utils/nodeIdErrorMessage.ts:24]: right that was my thought too. I like it more descriptive. and then if they have the length right, and it starts with NodeId then we check if it's a valid checksum

File: src/modules/components/Wizard/steps/WizardNodeID.tsx (hunk 1 above)
Comment (emersoncloud) [reply src/modules/components/Wizard/steps/WizardNodeID.tsx:92]: ya you're right great point

**Pull request merged**